service/src/main/java/org/broadinstitute/listener/relay/inspectors/RequestLogger.java

```diff
@@ -0,0 +1,124 @@
+package org.broadinstitute.listener.relay.inspectors;
+
+import com.microsoft.azure.relay.RelayedHttpListenerRequest;
+import java.io.IOException;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.TreeMap;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.listener.relay.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RequestLogger {
+
+  private final Logger logger = LoggerFactory.getLogger(RequestLogger.class);
+  private static final List<String> MUST_MASKED_HEADER_NAMES = List.of("Authorization", "Cookie");
+
+  /**
+   * Logs a relayed HTTP request with the result status code
+   *
+   * @param relayedHttpListenerRequest Relayed request
+   * @param statusCode Result of the request
+   * @param requestTimestamp Timestamp of the request
+   * @param prefix Logging prefix to include
+   * @throws IOException
+   * @throws InterruptedException
+   */
+  public void logRequest(
+      RelayedHttpListenerRequest relayedHttpListenerRequest,
+      int statusCode,
+      OffsetDateTime requestTimestamp,
+      String prefix)
+      throws IOException, InterruptedException {
+    if (relayedHttpListenerRequest == null) {
+      logger.warn("Null request provided for logging");
+      return;
+    }
+
+    // HTTP headers are case-insensitive
+    var headers = new TreeMap<String, String>(String.CASE_INSENSITIVE_ORDER);
+    headers.putAll(relayedHttpListenerRequest.getHeaders());
+
+    var referer = headers.getOrDefault("Referer", "");
+    var origin = headers.getOrDefault("Origin", "");
+    var ua = headers.getOrDefault("User-Agent", "");
+
+    String endpoint = "";
+    if (relayedHttpListenerRequest.getRemoteEndPoint() != null) {
+      endpoint = relayedHttpListenerRequest.getRemoteEndPoint().toString();
+    }
+    var claims = getTokenClaims(headers);
+
+    // log in a single apache-ish line
+    logger.info(
```

Comment:
All this just goes to stdout right, and is picked up by our collection rules? Was wondering if we need different appenders or something for "apache logs" vs other logs, but maybe not..

Reply:
It just goes into the container logs, i.e., you can query for it like so:

```ContainerLog
| where TimeGenerated > ago (48h)
| where ContainerID in ('ad2e08e477db3f9ee91868528344c169bdca9adb5aabfb416ef4e2a67aac5b4c')
| where LogEntry contains "RELAY_REQUEST_RESPONSE"
| order by TimeGenerated desc
| project LogEntry```

Reply:
It might be helpful to have in a JSON form too, but imo that'd blow up log volume unnecessarily, including due to how verbose it is. Workflows uses that in their logs, but I'm not aware of others.